x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedJobBuilder.java

```diff
@@ -101,8 +103,14 @@ void build(String datafeedId, ActionListener<DatafeedJob> listener) {
         );
 
         // Create data extractor factory
-        Consumer<DatafeedTimingStats> datafeedTimingStatsHandler = timingStats -> {
-            context.timingStatsReporter = new DatafeedTimingStatsReporter(timingStats, jobResultsPersister);
+        Consumer<DatafeedTimingStats> datafeedTimingStatsHandler = initialTimingStats -> {
+            DatafeedTimingStatsPersister timingStatsPersister = new DatafeedTimingStatsPersister() {
+                @Override
+                public void persistDatafeedTimingStats(DatafeedTimingStats timingStats, WriteRequest.RefreshPolicy refreshPolicy) {
+                    jobResultsPersister.persistDatafeedTimingStats(timingStats, refreshPolicy);
+                }
+            };
+            context.timingStatsReporter = new DatafeedTimingStatsReporter(initialTimingStats, timingStatsPersister);
```

Comment:
You might be able to pass here:

```
context.timingStatsReporter = new DatafeedTimingStatsReporter(initialTimingStats, jobResultsPersister::persistDatafeedTimingStats);
```

and then you won't need to create the `timingStatsPersister` object.
```

Reply:
Done.